cloudnet-modules/cloudnet-database-mongodb/src/main/java/de/dytanic/cloudnet/ext/database/mongodb/MongoDatabaseProvider.java

```diff
@@ -0,0 +1,149 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+@Getter
+public class MongoDatabaseProvider extends AbstractDatabaseProvider {
+
+    private MongoClient client;
+    private JsonDocument config;
+    private DatabaseConnectionData databaseConnectionData;
+    private ExecutorService executorService;
+
+    public MongoDatabaseProvider(JsonDocument config, ExecutorService executorService) {
+        this.config = config;
+        this.executorService = executorService;
+        this.databaseConnectionData = DatabaseConnectionData.fromJsonDocument(config);
+    }
+
+    public void closeSession() {
+        try {
+            getClient().close();
+        } catch (Exception e) {
+        }
+    }
+
+    public boolean isConnected() {
+        return getClient() != null;
+    }
+
+    @Override
+    public boolean init() throws Exception {
+        Executors.newScheduledThreadPool(1).scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                if (!isConnected()) {
+                    try {
+                        closeSession();
+                    } catch (Exception ex) {
+                    }
+                    connect();
+                }
+            }
+
+        }, 0, 3, TimeUnit.MINUTES);
+        return true;
+    }
+
+    public void connect() {
+        try {
+            this.client = MongoClients.create(new ConnectionString(this.databaseConnectionData.toConnectionString()));
+        } catch (Exception ex) {
+            CloudNetDriver.getInstance().getLogger().error(ex.getMessage(), ex);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "MongoDB-Provider";
+    }
+
+    public MongoDatabase getMongoDatabase() {
+        if (getClient() == null) {
+            connect();
+        }
+        return getClient().getDatabase(getDatabaseConnectionData().getDatabase());
+    }
+
+    @Override
+    public Database getDatabase(String name) {
+        return new de.dytanic.cloudnet.ext.database.mongodb.MongoDatabase(this, name, executorService);
+    }
+
+    @Override
+    public boolean containsDatabase(String name) {
+        return client.getDatabase(name) != null;
+    }
+
+    @Override
+    public @NotNull ITask<Boolean> containsDatabaseAsync(String name) {
+        return this.schedule(() -> containsDatabase(name));
+    }
+
+    @Override
+    public @NotNull ITask<Boolean> deleteDatabaseAsync(String name) {
+        return this.schedule(() -> deleteDatabase(name));
+    }
+
+    @Override
+    public @NotNull ITask<Collection<String>> getDatabaseNamesAsync() {
+        return this.schedule(() -> getDatabaseNames());
+    }
+
+    private <T> ITask<T> schedule(Callable<T> callable) {
+        ITask<T> task = new ListenableTask<>(callable);
+        this.executorService.execute(() -> {
+            try {
+                Thread.sleep(0, 100000);
```

Comment:
Why?

Reply:
Why you need that, this is a active thread wating and can be make concurrency problems!